internal/corazawaf/rule.go

```diff
@@ -210,9 +213,34 @@ func (r *Rule) doEvaluate(tx *Transaction) []types.MatchData {
 					// We could try for each until found
 					args, errs = r.executeTransformationsMultimatch(arg.Value())
 				} else {
-					ars, es := r.executeTransformations(arg.Value())
-					args = []string{ars}
-					errs = es
+					switch {
```

Comment:
I think 'doEvaluate' method is complex enough already so maybe we could extract this in another method?

Reply:
Yup extracted